### PR TITLE
Testing async blog post

### PR DESCRIPTION
**Pull request by alanvardy**

### CODE REVIEW

**Comment by alanvardy** [comment]
Current dependencies on/for this PR:
* main
  * **PR #72** <a href="https://app.graphite.dev/github/pr/alanvardy/alan_vardy/72" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/alanvardy/alan_vardy/72?utm_source=stack-comment).

File: posts/2022/09-15-testing-async.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+When running an async process in manual mode, any database calls it makes will error because there was no transaction taken out for that PID (only for the main test process). It would be easy to set `async: false` on the `use DataCase` macro to resolve the issue, but the trouble with setting these tests to synchronous is that it dramatically slows down the runtime of the test suite, and we just need to come back and fix it later anyways.
+
+The solution is to use the `caller` option of Repo. If we pass in the PID of the main test process it will use that PID when finding the transaction rather than its own.
+
+```elixir
+MyApp.Repo.all(from(u in User), caller: test_pid)
+```
```
Comment (CorneliaKelinske) [inline posts/2022/09-15-testing-async.md]: Where does this code live? in which file?

File: posts/2022/09-15-testing-async.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+When starting up the async process in `start_supervised!/2`, pass in the PID of the test process and use that for all repo calls. You can see an example of this in the last section.
```
Comment (CorneliaKelinske) [inline posts/2022/09-15-testing-async.md]: how do i get the test_pid?
Is it just:

setup do
  start_supervised!({SomeProcess, test_pid})
end

?

File: posts/2022/09-15-testing-async.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+  assert_receive(:sync, 500) in test file
+  """
+  if Mix.env() === :test do
+    def maybe_send_sync(pid), do: send(pid, :sync)
+  else
+    def maybe_send_sync(_pid), do: :sync
+  end
+```
+
```
Comment (CorneliaKelinske) [inline posts/2022/09-15-testing-async.md:96]: Where does this code live?

Also: might be worth showing where you use the message sending in a test.
I find the article super interesting, but I'm not sure if I would be readily able to test that way based on the description in this post.

Can you include a full example below the text? Where it is all put together?

So something like:
Filename:
```code```

filename:
```code```

If that makes sense?

**Review by CorneliaKelinske — CHANGES_REQUESTED** [review]
I find this article super interesting. It could be a bit clear and easier to follow.

**Comment by alanvardy** [comment]
Thank you Cornelia, I think I addressed your questions

**Review by CorneliaKelinske — APPROVED** [review]

**Pull request merged**